### DISCUSSION TITLE
Routes by decorators

### DISCUSSION THREAD

**Discussion by chiqui3d** [vercel/next.js#17127] [Ideas]
The only thing I miss in Next.js after working with big libraries and frameworks is the possibility to work with Typescript decorator. It is the most comfortable way I found on the roads after a long time.

An example would be this:

pages/blog/post.ts

```ts
  export class Post{

   @Router("/post/:slug")
   @Redirect("https://www.example.com/post/:slug")
   @Headers([])
   async postBySlug() {
    // 
   }

  }
```
The problem is that it would have to be with `class`, as `functional component` does not work with decorator, but you could use classes for the "pages" and functional components for the "components" 🙄.

I'd like to make a package about this, but looking at the route setup in Next.js I can't think of anything to do with this. Any ideas?

**Comment by leeuwis** [comment]
@chiqui3d we've created a library for the next api routes that's similar to the way you describe above: [next-api-decorators](https://github.com/storyofams/next-api-decorators). Maybe you like it and/or it could give you some inspiration if you want to do something similar for the pages/components.
